src/backend/optimizer/plan/aqumv.c

```diff
@@ -151,6 +151,9 @@ answer_query_using_materialized_views(PlannerInfo *root,
 		has_subclass(origin_rel_oid))
 		return mv_final_rel;
 
+	if (get_rel_relkind(origin_rel_oid) == RELKIND_FOREIGN_TABLE && !aqumv_allow_foreign_table)
```

Comment:
> create readable external table aqumv_ext_r ()

can we get `readable` from table  ?

Reply:
GP forbids  creating a matview as select from writable external table.

Reply:
> create readable external table aqumv_ext_r(id int) 
location ('demoprot://aqumvtextfile.txt') format 'text';

1. can we mantain `aqumvtextfile.txt`status in `gp_matview_aux`use file timestamp?

Reply:
It's just an example, they could be files,  remote servers or another databases outside CBDB. You couldn't know the status of them and pointless for that. See more in #693